### PR TITLE
va-radio, va-radio-option: Screen reader and keyboard navigation updates

### PR DESCRIPTION
**Pull request by powellkerry**
## Chromatic

https://2075-radio-group-accessibility--60f9b557105290003b387cd5.chromatic.com

## Description
- Remove tabIndex from radio options. They were causing options to be read as a "group"
- Update input Ids so they don't conflict with the web-component element id. This was causing the screen reader to not read the option label
- Change va-radio-option from `shadow: true` to `shadow: false`. This was causing the screen reader to not announce the total number of options and dictate that each option was its own radio group
- Remove label `id` attribute as this is not used and was causing AXE checks to complain about duplicate ids
- Update CSS to work with not having a host/shadow root
- Update tests for changes to HTML and functionality being tied to the input and not the web-component
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2075

## Testing done
- Local testing in Chrome, Safari, Firefox and Edge
- Screen reader testing done with Mac VoiceOver

## Screenshots
v1 Before:
Screenshot 2023-09-20 at 3 07 36 PM

v1 After:
Screenshot 2023-09-20 at 3 08 13 PM

v3 Before:
Screenshot 2023-09-20 at 3 08 44 PM

v3 After:
Screenshot 2023-09-20 at 3 46 02 PM

## Acceptance criteria
- [ ] Component works as before functionally
- [ ] Criteria found in card by @rsmithadhoc are met

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by powellkerry** [comment]
There is a difference in appearance between before and after v3 versions, working on hunting that down.

**Comment by powellkerry** [comment]
tabIndex was just introduced in the last release. We need to make sure that this change does not introduce a regression. See: https://github.com/department-of-veterans-affairs/component-library/pull/786

**Review by rsmithadhoc — CHANGES_REQUESTED** [review]
@powellkerry Thanks for picking this up, this worked perfectly for me with keyboard and screen reader. However, I left a comment for the focus style not using the VA yellow.

Thanks!

File: packages/web-components/src/components/va-radio-option/va-radio-option.scss [hunk 1]
```diff
@@ -14,33 +14,44 @@
...
-:host {
+va-radio-option {
   display: block;
   margin-top: 12px;
 }
 
-:host(:focus) {
+va-radio-option input:focus {
   outline: none !important;
 }
```
[the remark this answers is not in the document]
Comment (powellkerry) [reply packages/web-components/src/components/va-radio-option/va-radio-option.scss:24]: This should be fixed. Removing the shadow on the options caused a bunch of small things like this, prior to that the USWDS version was probably inheriting that style from the previous version.

**Review by rsmithadhoc — APPROVED** [review]
Looks good, thanks!

**Pull request merged**